Makefile

```
@@ -159,7 +159,10 @@ ARTIFACTS ?= $(ROOT_DIR)/_artifacts
 E2E_CONF_FILE ?= $(ROOT_DIR)/test/e2e/config/e2e_conf.yaml
 E2E_OUT_DIR ?= $(ROOT_DIR)/test/e2e/_out
 E2E_CONF_FILE_ENVSUBST ?= $(E2E_OUT_DIR)/$(notdir $(E2E_CONF_FILE))
-E2E_CONTAINERS ?= quay.io/metal3-io/cluster-api-provider-metal3 quay.io/metal3-io/baremetal-operator quay.io/metal3-io/ip-address-manager
+CAPM3_IMAGE ?= ${CONTAINER_REGISTRY}/metal3-io/cluster-api-provider-metal3:${CAPM3RELEASE:-latest}
```

Who initially defines CONTAINER_REGISTRY here?

In CAPM3, it is only mentioned in e2e_conf.yaml, so it is probably empty here, unless it was already defined in dev-env first. With CAPM3 e2e, the test cycle starts on CAPM3 side, not from dev-env, so does this actually work?

In this Makefile, registry variables defined are REGISTRY, STAGING_REGISTRY and PROD_REGISTRY?

---

CONTAINER_REGISTRY can be defined in project-infra (and is defined there for e2e-integration-tests) or gets a default value from dev-env. I can change `${CONTAINER_REGISTRY} `to `${CONTAINER_REGISTRY:quay.io}` just to be sure. 

I've been looking at [e2e-integration-test on release-1.9 ](https://jenkins.nordix.org/view/Metal3%20Periodic/job/metal3-periodic-ubuntu-e2e-integration-test-release-1-9/40/consoleFull) when making this change.

Here are some copied from the log I've linked above. Times are system time. As you can see from here, yes we starts from CAPM3 e2e but the tests very quickly moves to execute in dev-env. Prepulling of images happens there and we come back to CAPM3 later where the some images get pulled again with latest tag (which is not correct for release-1.9 tests). 
```
01:31:43  Running the tests
01:31:43  + make test-e2e
...
01:31:45  + pushd /opt/metal3-dev-env/metal3-dev-env
01:31:45  /opt/metal3-dev-env/metal3-dev-env ~/tested_repo
01:31:45  + make
01:31:45  make[1]: Entering directory '/opt/metal3-dev-env/metal3-dev-env'
...
01:37:25  ++ sudo docker pull registry.nordix.org/quay-io-proxy/metal3-io/baremetal-operator:v0.9.0
01:37:27  v0.9.0: Pulling from quay-io-proxy/metal3-io/baremetal-operator
...
01:56:45  + make e2e-tests
01:56:45  make[1]: Entering directory '/home/metal3ci/tested_repo'
...
01:56:50  Using default tag: latest
01:56:52  latest: Pulling from metal3-io/baremetal-operator
01:56:52  dd5ad9c9c29f: Already exists
01:56:52  960043b8858c: Already exists
01:56:52  b4ca4c215f48: Already exists
01:56:52  eebb06941f3e: Already exists
01:56:52  02cd68c0cbf6: Already exists
01:56:52  d3c894b5b2b0: Already exists
01:56:52  b40161cd83fc: Already exists
01:56:52  46ba3f23f1d3: Already exists
01:56:52  4fa131a1b726: Already exists
01:56:52  fc8b0dbaf292: Pulling fs layer
01:56:53  fc8b0dbaf292: Verifying Checksum
01:56:53  fc8b0dbaf292: Download complete
01:56:54  fc8b0dbaf292: Pull complete
```

---

Note that this is Makefile and not bash, so the variable syntax is not `{}` but `()`.

Also you need to make fallbacks differently.

```suggestion
CAPM3RELEASE ?= latest
CAPM3_IMAGE ?= $(CONTAINER_REGISTRY)/metal3-io/cluster-api-provider-metal3:$(CAPM3RELEASE)
```